aws-api-appsync/src/main/java/com/amplifyframework/api/aws/AppSyncGraphQLRequest.java

```diff
@@ -288,19 +297,23 @@ public Builder variable(@NonNull String key, String type, Object value) {
             Objects.requireNonNull(this.operation);
             Objects.requireNonNull(this.responseType);
 
-            if (modelClass != null) {
-                // Derive modelSchema and selectionSet from modelClass.
+            if (modelClass == null && modelSchema == null) {
```

Comment:
Not yet :-(

Reply:
It would seem so. But when I tried to carry this out, a bunch of tests broke in non-obvious ways. I'd like to flag this one and come back to it. I'll leave a TODO in the code.

Reply:
fair enough!  we can come back to it.